subsys/bluetooth/host/Kconfig.gatt

```diff
@@ -223,26 +223,26 @@ if BT_GAP_PERIPHERAL_PREF_PARAMS
 config BT_PERIPHERAL_PREF_MIN_INT
 	int "Peripheral preferred minimum connection interval in 1.25ms units"
 	default 24
-	range 6 65535
+	range 6 65535 if !BT_CONN_PARAM_ANY
 	help
 	  Range 3200 to 65534 is invalid. 65535 represents no specific value.
 
 config BT_PERIPHERAL_PREF_MAX_INT
 	int "Peripheral preferred maximum connection interval in 1.25ms units"
 	default 40
-	range 6 65535
+	range 6 65535 if !BT_CONN_PARAM_ANY
 	help
 	  Range 3200 to 65534 is invalid. 65535 represents no specific value.
 
 config BT_PERIPHERAL_PREF_LATENCY
 	int "Peripheral preferred peripheral latency in Connection Intervals"
 	default 0
-	range 0 499
+	range 0 499 if !BT_CONN_PARAM_ANY
 
 config BT_PERIPHERAL_PREF_TIMEOUT
 	int "Peripheral preferred supervision timeout in 10ms units"
 	default 42
-	range 10 65535
+	range 10 65535 if !BT_CONN_PARAM_ANY
```

Comment:
Outside the scope of this PR, but the max values of `BT_PERIPHERAL_PREF_MIN_INT`, `BT_PERIPHERAL_PREF_MAX_INT` and `BT_PERIPHERAL_PREF_TIMEOUT` are actually above the values from the HCI spec, and we should consider to set them to more sane values :)